server/mqtt_test.go

```diff
@@ -4152,6 +4179,39 @@ func TestMQTTPublishRetain(t *testing.T) {
 	}
 }
 
+func TestMQTTQoS2RetainedReject(t *testing.T) {
+	o := testMQTTDefaultOptions()
+	s := testMQTTRunServer(t, o)
+	mc1, rs1 := testMQTTConnect(t, &mqttConnInfo{cleanSess: true}, o.MQTT.Host, o.MQTT.Port)
+	testMQTTCheckConnAck(t, rs1, mqttConnAckRCConnectionAccepted, false)
+	testMQTTPublish(t, mc1, rs1, 2, false, true, "foo", 1, []byte("qos2 failed"))
+	testMQTTPublish(t, mc1, rs1, 1, false, true, "bar", 2, []byte("qos1 retained"))
+	testMQTTFlush(t, mc1, nil, rs1)
+	testMQTTDisconnect(t, mc1, nil)
+	mc1.Close()
+	s.Shutdown()
+
+	// restart the server with MQTT disabled; we should be using the same
```

Comment:
Not sure the comment is related to what we do here?

Reply:
Added/clarified the comment, s/MQTT/MQTT QoS2/